recipe/meta.yaml

```
@@ -0,0 +1,67 @@
+{% set name = "momepy" %}
+{% set version = "0.7.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/momepy-{{ version }}.tar.gz
+  sha256: 413ed9221dc4b2190621f8cb37789f09ea882f7e81dfbffd5731a22da6e61bab
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+  skip: True  # [py<310 or s390x]
+
+requirements:
+  host:
+    - python
+    - setuptools >=61.0
+    - setuptools-scm >=6.2
+    - pip
+  run:
+    - python
+    - geopandas >=0.12.0
+    - libpysal >=4.6.0
+    - networkx >=2.7
+    - packaging
+    - pandas !=1.5.0
+    - shapely >=2
+    - tqdm >=4.63.0
+    - inequality
+    - jupyter
+    - mapclassify
+    - matplotlib
+    - networkx
+    - numba
+    - esda
+
+test:
+  source_files:
+    - ./momepy/tests
+  imports:
+    - momepy
+  commands:
+    #- pip check
```

```suggestion
    - pip check
```
Maybe it was failing because of the missing wheel?

---

Added `wheel` as a host requirement, but still having the same failure.

---

```
+ pip check
momepy 0.7.0 has requirement libpysal>=4.6.0, but you have libpysal 0.0.0.
WARNING: Tests failed for momepy-0.7.0-py312hd43f75c_0.tar.bz2 - moving package to /opt/conda/conda-bld/broken
TESTS FAILED: momepy-0.7.0-py312hd43f75c_0.tar.bz2
```

---

Ohhh, it's not finding your `libpysal`

---

Are you only getting that error locally?

---

If so, try running `conda build <feedstock> -c <staging_channel>`

---

I am getting it locally, currently running `conda build -c https://staging.continuum.io/prefect/fs/libpysal-feedstock/pr4/eef6b13 momepy-feedstock/`

---

I want to say the order matters, but I'm not 100% sure.

---

Nvm! The order doesn't matter! I just verified. Lol

---

In libpysal you need to set SETUPTOOLS_SCM_PRETEND_VERSION. (Setuptools scm is not able to find a version because the build is not from a git repo. So it sets 0.0.0. This can be overriden. See doc https://setuptools-scm.readthedocs.io/en/latest/overrides/ and example: https://github.com/AnacondaRecipes/menuinst-feedstock/blob/master/recipe/meta.yaml#L14-L15